src/OrchardCore.Modules/OrchardCore.Media/Controllers/AdminController.cs

```diff
@@ -201,7 +201,7 @@ public async Task<IActionResult> Upload(string path, string extensions)
                 {
                     var extension = Path.GetExtension(file.FileName);
 
-                    if (!allowedExtensions.Contains(extension))
+                    if (!allowedExtensions.Contains(extension, StringComparer.OrdinalIgnoreCase))
```

Comment:
Thanks for your contribution, but this doesn't look correct. `allowedExtensions` is returned by `GetRequestedExtensions` and it is a case insensitive `HashSet` already. You're also replacing the `HashSet.Contains` method of with an extension method of `IEnumerable`, which is bad for perf.
What are you trying to solve?

Reply:
If i try to select a media named "articlephoto.JPG" for a MediaField, when i save the contentitem i get this error:
"Media extension is not allowed. Only media with '.jpg, .jpeg' extensions are allowed."
If i rename the file in lower case, everything works correctly.

Reply:
I agree with @gvkries contains() in this case is already case insensate. I tested uploading an img with`.JPG` locally and it works.

Navigate to Admin/Media/Options?admin=579832242 and make sure `.jpg` is listed in the Allowed file extensions.

Reply:
.jpg is listed in "Allowed file extensions"
in the media field options of my contentitem i limited the file extensions only to ".jpg and .jpeg"
i can upload the image but when i publish the contentitem i get this error

![image](https://github.com/user-attachments/assets/8c583b51-3405-43c3-b202-c9738ebaa1fa)

Reply:
Which Version of Orchard are you using?

Reply:
2.1.2

Reply:
Thanks for the explanation. I don't think this fixes the issue, the issue is more likely located in the `MediaFieldDisplayDriver`:
https://github.com/OrchardCMS/OrchardCore/blob/aa9fdcbf49774a0ab96819c6e47db67928fe746b/src/OrchardCore.Modules/OrchardCore.Media/Drivers/MediaFieldDisplayDriver.cs#L121

Also, please open an issue first.

Reply:
I can't reproduce the problem on the main branch. Can you share the image file you are trying to upload? I am wondering if there is something with with that file.

Reply:
So yes, I was able to reproduce it with media field settings...

Reply:
Thanks, now please move your fix to the media field display driver and we're ready to take it! We're always very happy about new contributors 🤗